### PR TITLE
feat(sdk-coin-atom): add support to serialise unsigned txn

### PR DESCRIPTION
**Pull request by dpkjnr**
Ticket: BG-67811

### CODE REVIEW

**Comment by dpkjnr** [comment]
unit tests are on the way

**Review by ewangbitgo — DISMISSED** [review]

File: modules/sdk-coin-atom/src/lib/iface.ts [hunk 1]
```diff
@@ -14,9 +14,9 @@ export interface MessageData {
   };
 }
 
-export interface GasFeeLimitData {
+export interface FeeData {
   amount: Coin[];
-  gas: number; // gas limit
+  gasLimit: number;
```
Comment (sachushaji) [inline modules/sdk-coin-atom/src/lib/iface.ts:19]: Seems to be breaking change?, Can you kindly mention in the commit message ([format](https://www.conventionalcommits.org/en/v1.0.0/#:~:text=in%20Semantic%20Versioning\).-,BREAKING,-CHANGE%3A%20a))

File: modules/sdk-coin-atom/src/lib/utils.ts [hunk 2]
```diff
@@ -166,13 +193,21 @@ export class Utils implements BaseUtils {
     return encodedMessage.typeUrl;
   }
 
-  getGasBudgetFromDecodedTx(decodedTx: DecodedTxRaw): GasFeeLimitData {
+  getGasBudgetFromDecodedTx(decodedTx: DecodedTxRaw): FeeData {
```
Comment (sachushaji) [inline modules/sdk-coin-atom/src/lib/utils.ts:196]: Kindly add jsdocs, lets be consistent with rest of the functions

File: modules/sdk-coin-atom/src/lib/utils.ts [hunk 3]
```diff
@@ -166,13 +193,21 @@ export class Utils implements BaseUtils {
...
     return {
       amount: decodedTx.authInfo.fee?.amount as Coin[],
-      gas: Number(decodedTx.authInfo.fee?.gasLimit),
+      gasLimit: Number(decodedTx.authInfo.fee?.gasLimit),
     };
   }
 
+  getPublicKeyFromDecodedTx(decodedTx: DecodedTxRaw): string | undefined {
```
Comment (sachushaji) [inline modules/sdk-coin-atom/src/lib/utils.ts:203]: Kindly add jsdocs, lets be consistent with rest of the functions, Kindly add to all functions i the file which doesn't have it

File: modules/sdk-coin-atom/src/atom.ts [hunk 4]
```diff
@@ -60,28 +62,82 @@ export class Atom extends BaseCoin {
     return 'ecdsa';
   }
 
-  verifyTransaction(params: VerifyTransactionOptions): Promise<boolean> {
-    throw new Error('Method not implemented.');
+  async verifyTransaction(params: VerifyTransactionOptions): Promise<boolean> {
```
Comment (sachushaji) [inline modules/sdk-coin-atom/src/atom.ts:65]: Didn't see a unit test for this function?

File: modules/sdk-coin-atom/src/atom.ts [hunk 5]
```diff
@@ -60,28 +62,82 @@ export class Atom extends BaseCoin {
...
+    return true;
   }
 
   async isWalletAddress(params: VerifyAddressOptions): Promise<boolean> {
     throw new Error('Method not implemented.');
   }
 
-  parseTransaction(params: ParseTransactionOptions): Promise<ParsedTransaction> {
-    throw new Error('Method not implemented.');
+  async parseTransaction(params: ParseTransactionOptions & { txHex: string }): Promise<ParsedTransaction> {
```
Comment (sachushaji) [inline modules/sdk-coin-atom/src/atom.ts:99]: Didn't see a unit test for this function?

File: modules/sdk-coin-atom/src/lib/utils.ts [hunk 6]
```diff
@@ -55,7 +69,21 @@ export class Utils implements BaseUtils {
    * @returns {boolean} - the validation result
    */
   isValidRawTransaction(rawTransaction: string): boolean {
```
Comment (sachushaji) [inline modules/sdk-coin-atom/src/lib/utils.ts:71]: Didn't see a unit test for this function?

**Review by sachushaji — CHANGES_REQUESTED** [review]
Thanks @dpkjnr for the work, Added some suggestions, Kindly add more unit tests for the new additions

File: modules/sdk-coin-atom/src/lib/utils.ts (hunk 2 above)
Comment (dpkjnr) [reply modules/sdk-coin-atom/src/lib/utils.ts:226]: done

File: modules/sdk-coin-atom/src/lib/utils.ts (hunk 3 above)
Comment (dpkjnr) [reply modules/sdk-coin-atom/src/lib/utils.ts:238]: done

File: modules/sdk-coin-atom/src/lib/transaction.ts [hunk 7]
```diff
@@ -116,26 +115,31 @@ export class Transaction extends BaseTransaction {
    *
    * @param rawTransaction
```
Comment (sachushaji) [inline modules/sdk-coin-atom/src/lib/transaction.ts:116]: Lets be consistent on jdocs
```suggestion
   * @param {string} rawTransaction
```

Comment (sachushaji) [reply modules/sdk-coin-atom/src/lib/transaction.ts:116]: NIt: Also if you can make the other functions in the file consistent with JSDocs that would be awesome

File: modules/sdk-coin-atom/src/lib/utils.ts [hunk 8]
```diff
@@ -1,47 +1,66 @@
...
   }
 
   /** @inheritdoc */
   isValidTransactionId(txId: string): boolean {
     return this.validateBlake2b(txId);
   }
 
+  /**
+   * Checks if transaction hash is in valid black2b format
+   */
```
Comment (sachushaji) [inline modules/sdk-coin-atom/src/lib/utils.ts:63]:
```suggestion
  /**
   * Checks if transaction hash is in valid black2b format
   * @param {string} hash
   * @returns {boolean}
   */
```

File: modules/sdk-coin-atom/src/lib/utils.ts [hunk 9]
```diff
@@ -54,16 +73,42 @@ export class Utils implements BaseUtils {
...
+      return false;
+    } catch (e) {
+      return false;
+    }
   }
 
   /**
    * Check the raw transaction has a valid format in the blockchain context, throw otherwise.
    *
    * @param {string} rawTransaction - Transaction in base64 string  format
```
Comment (sachushaji) [inline modules/sdk-coin-atom/src/lib/utils.ts:109]:
```suggestion
   * @param {string | undefined} rawTransaction - Transaction in base64 string  format
```

File: modules/sdk-coin-atom/src/lib/utils.ts [hunk 10]
```diff
@@ -72,15 +117,17 @@ export class Utils implements BaseUtils {
     }
   }
 
+  /**
+   * Validates if the address matches with regex @see accountAddressRegex
+   *
+   * @param address
```
Comment (sachushaji) [inline modules/sdk-coin-atom/src/lib/utils.ts:123]:
```suggestion
   * @param {string} address
```

File: modules/sdk-coin-atom/src/lib/utils.ts [hunk 11]
```diff
@@ -94,6 +141,12 @@ export class Utils implements BaseUtils {
     return true;
   }
 
+  /**
+   * Validates whether amount is in range
+   *
+   * @param amount
+   * @returns
+   */
```
Comment (sachushaji) [inline modules/sdk-coin-atom/src/lib/utils.ts:149]:
```suggestion
   * Validates whether amount is in range
   *
   * @param {number}  amount
   * @returns {boolean}
   */
```

File: modules/sdk-coin-atom/src/lib/utils.ts (hunk 2 above)
Comment (sachushaji) [reply modules/sdk-coin-atom/src/lib/utils.ts:226]: Can you a decorate the params with type also and mention the return type too, Kindly do that for the rest of the functions also if you can

**Review by sachushaji — APPROVED** [review]
Thanks for adding more unit tests, I have also added some more NITs, mainly on keeping the jsdocs a bit consistent. You can take it up in a follow up PR.

**Review by mukeshsp — APPROVED** [review]

**Pull request merged**